### PR TITLE
hipchat alerter

### PR DESCRIPTION
**Pull request by doodyparizada**
send matches to a room in hipchat

@rounds/server

### CODE REVIEW

File: elastalert/alerts.py [hunk 1]
```diff
@@ -443,3 +444,41 @@ def alert(self, matches):
     def get_info(self):
         return {'type': 'command',
                 'command': ' '.join(self.last_command)}
+
+
+class HipChatAlerter(Alerter):
+
+    """ Send notfication to a HipChat room. """
```
Comment (oryband) [inline elastalert/alerts.py]: no need for spaces

File: elastalert/alerts.py [hunk 2]
```diff
@@ -443,3 +444,41 @@ def alert(self, matches):
...
+            'color': self.alert_color,
+            'message_format': 'html',
+            'notify': False,
+            'message': str(message),
+        }
+        headers = {'content-type': 'application/json'}
+        requests.post(uri,
+                      params={'auth_token': self.auth_token},
+                      data=payload,
+                      headers=headers)
```
Comment (oryband) [inline elastalert/alerts.py]: headers can be written inline and save the variable decl, same for payload

File: elastalert/alerts.py [hunk 3]
```diff
@@ -443,3 +444,41 @@ def alert(self, matches):
...
+        """ post a message in a hipchat room. """
+        uri = '%s/room/%s/notification' % (self.base_uri, room_id)
+        payload = {
+            'color': self.alert_color,
+            'message_format': 'html',
+            'notify': False,
+            'message': str(message),
+        }
+        headers = {'content-type': 'application/json'}
+        requests.post(uri,
```
Comment (oryband) [inline elastalert/alerts.py]: use a request's session object, cached as a class member
take in account that this check will repeat itself lots of times

Comment (oryband) [inline elastalert/alerts.py]: though i'm not sure we want to close the socket
i dunno
@fuzzyami @bolshoy @ofir-petrushka wdyt?

Comment (bolshoy) [inline elastalert/alerts.py]: @oryband this is http, so socket closing depends on several things. Donno what this implementation does, but generally if you do not supply "Connection: close" header (and do not disable "Expect: 100-continue" headers if the request is large), by default the connection should linger for some predefined time.

File: elastalert/alerts.py (hunk 2 above)
Comment (liorama) [inline elastalert/alerts.py]: but this is much more readable.

File: elastalert/alerts.py (hunk 1 above)
Comment (doodyparizada) [inline elastalert/alerts.py]: I know, but I'm keeping with the current style in this file.

File: elastalert/alerts.py (hunk 3 above)
Comment (doodyparizada) [inline elastalert/alerts.py]: KISS

File: elastalert/alerts.py (hunk 2 above)
Comment (doodyparizada) [inline elastalert/alerts.py]: I might do the opposite and pull out the params into a separate var too...

Comment (liorama) [inline elastalert/alerts.py]: https://www.youtube.com/watch?v=JoqDYcCDOTg

**Comment by ofir-petrushka** [comment]
Was this converted to slack?

**Comment by bolshoy** [comment]
@ofir-petrushka  another branch with hipchat/slack already merged to Yelp/master

**Comment by ofir-petrushka** [comment]
@bolshoy @doodyparizada delete/close this then?

**Pull request closed without merging**